letsencrypt/cli.py

```diff
@@ -198,81 +199,79 @@ def _find_duplicative_certs(domains, config, renew_config):
-            ).format(identical_names_cert.configfile.filename)
+            ).format(ident_names_cert.configfile.filename, br=os.linesep)
         elif subset_names_cert is not None:
             question = (
                 "You have an existing certificate that contains a portion of "
-                "the domains you requested (ref: {0})\n\nIt contains these "
-                "names: {1}\n\nYou requested these names for the new "
-                "certificate: {2}.\n\nDo you want to replace this existing "
+                "the domains you requested (ref: {0}){br}{br}It contains these "
+                "names: {1}{br}{br}You requested these names for the new "
+                "certificate: {2}.{br}{br}Do you want to replace this existing "
                 "certificate with the new certificate?"
             ).format(subset_names_cert.configfile.filename,
                      ", ".join(subset_names_cert.names()),
-                     ", ".join(domains))
+                     ", ".join(domains),
+                     br=os.linesep)
         if question is None:
             # We aren't in a duplicative-names situation at all, so we don't
             # have to tell or ask the user anything about this.
             pass
-        elif zope.component.getUtility(interfaces.IDisplay).yesno(
-                question, "Replace", "Cancel"):
-            treat_as_renewal = True
+        elif config.no_confirm or zope.component.getUtility(
+                interfaces.IDisplay).yesno(question, "Replace", "Cancel"):
+            renewal = True
         else:
             reporter_util = zope.component.getUtility(interfaces.IReporter)
             reporter_util.add_message(
                 "To obtain a new certificate that {0} an existing certificate "
                 "in its domain-name coverage, you must use the --duplicate "
-                "option.\n\nFor example:\n\n{1} --duplicate {2}".format(
-                    "duplicates" if identical_names_cert is not None else
-                    "overlaps with", sys.argv[0], " ".join(sys.argv[1:])),
+                "option.{br}{br}For example:{br}{br}{1} --duplicate {2}".format(
+                    "duplicates" if ident_names_cert is not None else
+                    "overlaps with",
+                    sys.argv[0], " ".join(sys.argv[1:]),
+                    br=os.linesep
+                ),
                 reporter_util.HIGH_PRIORITY)
-            return 1
+            raise errors.Error(
+                "User did not use proper CLI and would like "
+                "to reinvoke the client.")
 
-    # Attempting to obtain the certificate
-    # TODO: Handle errors from _init_le_client?
-    le_client = _init_le_client(args, config, authenticator, installer)
-    if treat_as_renewal:
-        lineage = identical_names_cert if identical_names_cert is not None else subset_names_cert
-        # TODO: Use existing privkey instead of generating a new one
```

Comment:
Why not generate a new key?  (Both are just symlinks)

I guess I feared a quick fix.  The correct fix should be to use the same key iff the requested keysize is equivalent to the key used previously.  Still... this leads to further complications.  

Say you revoked the certificate with the previous key because of something like Heartbleed, or better yet, you intend to revoke the certificate but will do so after you have a new certificate in hand.  Then you have a certificate with the same key... and you probably don't even realize it.

I don't immediately see any problems with using a new key and it simplifies our lives greatly.